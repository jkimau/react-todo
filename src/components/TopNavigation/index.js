import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { topMenuHeight } from 'global/variables';

const Nav = styled.nav`
  height: ${topMenuHeight};
`;

const List = styled.li`
  list-style: none;
  display: inline-block;
  padding: 0 5px;
`;

export default class TopNavigation extends Component {
  constructor(props) {
    super(props);

    this.changeViewMode = this.changeViewMode.bind(this);
  }

  changeViewMode(mode) {
    this.props.changeViewModeHandler(mode);
  }

  render() {
    const linkStyle = { textDecoration: 'none' };
    return (
      <Nav>
        <ul>
          <List>
            <Link
              id="list-nav-link"
              style={{
                ...linkStyle,
                color: this.props.viewMode === 'list' ? 'blue' : 'black'
              }}
              to='/list'
              onClick={() => {this.changeViewMode('list')}}>
              List view
            </Link>
          </List>
          <List>
            <Link
              id="calendar-nav-link"
              style={{
                ...linkStyle,
                color: this.props.viewMode === 'calendar' ? 'blue' : 'black'
              }}
              to='/calendar'
              onClick={() => {this.changeViewMode('calendar')}}>
              Calendar view
            </Link>
          </List>
        </ul>
      </Nav>
    )
  }
}

TopNavigation.propTypes = {
  viewMode: PropTypes.string,
  changeViewModeHandler: PropTypes.func
};
