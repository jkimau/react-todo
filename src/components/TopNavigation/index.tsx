import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

import { topMenuHeight } from 'global/variables';

const Nav = styled.nav`
  width: 40%;
  height: ${topMenuHeight};
`;

const List = styled.li`
  list-style: none;
  display: inline-block;
  padding: 0 5px;
`;

interface TopNavigationPropTypes {
  viewMode: string,
  changeViewModeHandler(mode: string): void
}

export default class TopNavigation extends Component<TopNavigationPropTypes, null> {
  constructor(props) {
    super(props);

    this.changeViewMode = this.changeViewMode.bind(this);
  }

  changeViewMode(mode: string) {
    this.props.changeViewModeHandler(mode);
  }

  render() {
    const linkStyle = { textDecoration: 'none' };
    return (
      <Nav>
        <ul>
          <List>
            <Link
              id="list-nav-link"
              style={{
                ...linkStyle,
                color: this.props.viewMode === 'list' ? 'blue' : 'black'
              }}
              to='/list'
              onClick={() => {this.changeViewMode('list')}}>
              List view
            </Link>
          </List>
          <List>
            <Link
              id="calendar-nav-link"
              style={{
                ...linkStyle,
                color: this.props.viewMode === 'calendar' ? 'blue' : 'black'
              }}
              to='/calendar'
              onClick={() => {this.changeViewMode('calendar')}}>
              Calendar view
            </Link>
          </List>
        </ul>
      </Nav>
    )
  }
}
