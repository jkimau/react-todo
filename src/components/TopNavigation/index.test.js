import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom'

import TopNavigation from '../TopNavigation';

describe('<TopNavigation/>', () => {
  const defaultProps = {
    viewMode: 'list',
    changeViewModeHandler: jest.fn()
  }

  const compileComponent = (props = defaultProps) => {
    return shallow(<TopNavigation {...props}/>);
  }

  test('snapshot - list viewMode', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <TopNavigation {...defaultProps}/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('snapshot - calendar viewMode', () => {
    const newProps = Object.assign(
      defaultProps,
      { viewMode: 'calendar' }
    );
    const tree = renderer.create(
      <MemoryRouter>
        <TopNavigation {...newProps}/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call changeViewMode() function', () => {
    const wrapper = compileComponent();
    wrapper.find('#list-nav-link').simulate('click');

    expect(wrapper.instance().props.changeViewModeHandler).toHaveBeenCalledWith('list');

    wrapper.find('#calendar-nav-link').simulate('click');

    expect(wrapper.instance().props.changeViewModeHandler).toHaveBeenCalledWith('calendar');
  });
});
