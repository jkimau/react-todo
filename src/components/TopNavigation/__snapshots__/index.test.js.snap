exports[`<TopNavigation/> snapshot - calendar viewMode 1`] = `
<nav
  className="irUjdh">
  <ul>
    <li
      className="gxaXoS">
      <a
        href="/list"
        id="list-nav-link"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }>
        List view
      </a>
    </li>
    <li
      className="gxaXoS">
      <a
        href="/calendar"
        id="calendar-nav-link"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
            "textDecoration": "none",
          }
        }>
        Calendar view
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<TopNavigation/> snapshot - list viewMode 1`] = `
<nav
  className="irUjdh">
  <ul>
    <li
      className="gxaXoS">
      <a
        href="/list"
        id="list-nav-link"
        onClick={[Function]}
        style={
          Object {
            "color": "blue",
            "textDecoration": "none",
          }
        }>
        List view
      </a>
    </li>
    <li
      className="gxaXoS">
      <a
        href="/calendar"
        id="calendar-nav-link"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }>
        Calendar view
      </a>
    </li>
  </ul>
</nav>
`;
